### PR TITLE
Migrate RustTest to Kotlin

### PR DESCRIPTION
**Pull request by oyeraghib**
## Purpose / Description
Migrate RustTest to Kotlin

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by Akshay0701 — APPROVED** [review]
LGTM, Thanks

**Comment by Akshay0701** [comment]
code looks good, but before marking it approved, please add a proper name to your commits

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/RustTest.kt [hunk 1]
```diff
@@ -13,38 +13,34 @@
...
+    fun collectionIsVersion11AfterOpen() {
         // This test will be decommissioned, but before we get an upgrade strategy, we need to ensure we're not upgrading the database.
-        String path = Shared.getTestFilePath(getTestContext(), "initial_version_2_12_1.anki2");
-        Collection collection = Storage.Collection(getTestContext(), path);
-        int ver = collection.getDb().queryScalar("select ver from col");
-        MatcherAssert.assertThat(ver, is(11));
+        val path = Shared.getTestFilePath(testContext, "initial_version_2_12_1.anki2")
+        val collection = Storage.Collection(testContext, path)
+        val ver = collection.db.queryScalar("select ver from col")
+        assertThat(ver, `is`(11))
```
Comment (david-allison) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/RustTest.kt:44]: keep `MatcherAssert.` here

**Review by david-allison — COMMENTED** [review]
nit, then good to go

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/RustTest.kt (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/RustTest.kt:44]: @oyeraghib fix this then can merge!

Comment (oyeraghib) [reply AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/RustTest.kt]: > @oyeraghib fix this then can merge!

done!!

**Review by mikehardy — APPROVED** [review]
Nice! Thanks

**Pull request merged**